include/RAJA/policy/cuda/launch.hpp

```diff
@@ -74,21 +74,38 @@ struct LaunchExecute<RAJA::cuda_launch_t<async, 1>> {
     if ( gridSize.x  > zero && gridSize.y  > zero && gridSize.z  > zero &&
          blockSize.x > zero && blockSize.y > zero && blockSize.z > zero ) {
 
+
       RAJA_FT_BEGIN;
       {
         //
         // Privatize the loop_body, using make_launch_body to setup reductions
         //
         BODY body = RAJA::cuda::make_launch_body(gridSize, blockSize, params.shared_mem_size, cuda_res, std::forward<BODY_IN>(body_in));
 
+        //
+        // Configure plugins
+        //
+        util::PluginContext context{util::make_context<RAJA::cuda_launch_t<async,1>>()};
+        util::callPreCapturePlugins(context);
+
+        using RAJA::util::trigger_updates_before;
+        auto p_body = trigger_updates_before(body);
+
+        util::callPostCapturePlugins(context);
+
+        util::callPreLaunchPlugins(context);
+
         //
         // Launch the kernel
         //
-        void *args[] = {(void*)&body};
+        void *args[] = {(void*)&p_body};
         RAJA::cuda::launch((const void*)func, gridSize, blockSize, args, params.shared_mem_size, cuda_res, async, kernel_name);
+
+        util::callPostLaunchPlugins(context);
```

Comment:
@davidbeckingsale @MrBurmark if you guys have a second, does this look like the correct integration of the plugins with launch? I can confirm this works with CHAI now.

Reply:
Looks good to me.

Reply:
Is there a generic launch in front of this where this could go? It's being repeated many times if its put here in the backend files.

Reply:
Yes, good idea. I'll investigate.